File: Php/tests/AppTests/Cashgame/Action/CheckpointModelTests.php
namespace tests\AppTests\Cashgame\Action{
	use entities\Cashgame;
	use entities\Homegame;
	use entities\Player;
	use tests\UnitTestCase;
	use app\Cashgame\Action\CheckpointModel;
	use DateTime;
	use entities\Checkpoints\ReportCheckpoint;
	use tests\TestHelper;

	class CheckpointModelTests extends UnitTestCase {

		private $checkpoint;
		private $isManager;

		function setUp(){
			checkpoint = new ReportCheckpoint(new DateTime('2010-01-01 01:00:00'), 200);
			isManager = false;
		}

		function getSut(){
			$cashgame = new Cashgame();
			$cashgame.setStartTime(new DateTime());
			$player = new Player();
			return new CheckpointModel(new Homegame(), $cashgame, $player, checkpoint, isManager);
		}

		function test_Timestamp_IsSet(){
			$sut = getSut();

			assertIdentical($sut.timestamp, '01:00');
		}

		function test_Description_IsSet(){
			$sut = getSut();

			assertIdentical($sut.description, 'Report');
		}

		function test_Stack_IsSet(){
			$sut = getSut();

			assertIdentical($sut.stack, '$200');
		}

		function test_ShowLink_NormalUser_IsFalse(){
			$sut = getSut();

			assertFalse($sut.showLink);
		}

		function test_ShowLink_ManagerUser_IsTrue(){
			isManager = true;

			$sut = getSut();

			assertTrue($sut.showLink);
		}

		function test_EditUrl_IsCorrectType(){
			$sut = getSut();

			assertIsA($sut.editUrl, 'app\Urls\CashgameCheckpointDeleteUrlModel');
		}

	}

}

File: Web.Tests/ModelTests/CashgameModels/CheckpointModelTests.cs
using System;
using Core.Classes;
using Core.Classes.Checkpoints;
using NUnit.Framework;
using Tests.Common;
using Web.Models.CashgameModels.Action;
using Web.Models.UrlModels;

namespace Web.Tests.ModelTests.CashgameModels.Action{
	
	public class CheckpointModelTests : WebMockContainer {

		private Checkpoint _checkpoint;
		private Role _role;

        [SetUp]
		public void SetUp(){
			_checkpoint = new Checkpoint{Timestamp = DateTime.Parse("2010-01-01 01:00:00"), Stack = 200};
			_role = Role.Player;
		}

        [Test]
		public void Timestamp_IsSet(){
			var sut = GetSut();

			Assert.AreEqual(sut.Timestamp, "01:00");
		}

		[Test]
		public void Description_IsSet(){
			var sut = GetSut();

			Assert.AreEqual(sut.Description, "Report");
		}

		[Test]
		public void Stack_IsSet(){
			var sut = GetSut();

			Assert.AreEqual(sut.Stack, "$200");
		}

		[Test]
		public void ShowLink_NormalUser_IsFalse(){
			var sut = GetSut();

			Assert.IsFalse(sut.ShowLink);
		}

		[Test]
		public void ShowLink_ManagerUser_IsTrue(){
			_role = Role.Manager;

			var sut = GetSut();

			Assert.IsTrue(sut.ShowLink);
		}

		[Test]
		public void EditUrl_IsCorrectType(){
			var sut = GetSut();

            Assert.IsInstanceOf<CashgameCheckpointDeleteUrlModel>(sut.EditUrl);
		}

        private CheckpointModel GetSut(){
			var cashgame = new Cashgame {StartTime = new DateTime()};
            var player = new Player();
			return new CheckpointModel(new Homegame(), cashgame, player, _checkpoint, _role);
		}

	}

}